hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java

```diff
@@ -422,4 +423,25 @@ public JobConfig setClassLoaderFactory(@Nullable JobClassLoaderFactory classLoad
     public JobClassLoaderFactory getClassLoaderFactory() {
         return classLoaderFactory;
     }
+
+    /**
+     * TODO [viliam] finish javadoc
+     * @return
+     */
+    public String getInitialSnapshotName() {
+        return initialSnapshotName;
+    }
+
+    /**
+     * TODO [viliam] finish javadoc
+     *
+     * You can restore state also for non-snapshotted jobs.
+     *
+     * @param initialSnapshotName
+     * @return
+     */
+    public JobConfig setInitialSnapshotName(String initialSnapshotName) {
```

Comment:
here we call it "initial snapshot" but in the Job API it's called `exportedState`. It doesn't quite match in my opinion.

Reply:
A logical name would be `importSnapshot`

Reply:
The export method is now called `exportSnapshot` and the holder object is `JobStateSnapshot`. I don't like `importSnapshot`, because we don't import anything. The initial snapshot can even be loaded multiple times, if the job restarts before it creates first regular snapshot. It is the snapshot we load initially if we don't have a regular snapshot.